tests/test_agent.py

```diff
@@ -1352,6 +1352,34 @@ def test_double_capture_run_messages() -> None:
         ]
     )
 
+@pytest.mark.parametrize('return_value,valid', [
+    ('', True),  
+    (None, False),
+    (5, True),  ## TODO - decide if this is valid
```

Comment:
Either intentionally or accidentally returning a number seems quite unlikely but I decided to cast to str and add a test.